src/dataframe/dataframe.jl

```diff
@@ -740,20 +740,28 @@ end
 ##############################################################################
 
 function categorical!(df::DataFrame, cname::Union{Integer, Symbol})
-    df[cname] = CategoricalVector(df[cname])
+    if Nulls.T(eltype(df[cname])) <: AbstractString
```

Comment:
Instead of adding this deprecation here, I'd rather do that directly in CategoricalArrays. That way we have only one deprecation to remove once we decide the deprecation period is over.